H5Timing.h

```diff
@@ -0,0 +1,45 @@
+#ifndef H5_TIMING_H
+
+#define H5_TIMING_H
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#undef H5_TIMING_ENABLE
+
+#ifdef H5_TIMING_ENABLE
```

Comment:
Wouldn't this be better to be controlled via cmake options? @knoepfel @sabasehrish do either of you know how that could be done?

Reply:
I can move it to CMAKE.

Reply:
@Dr15Jones, @QiaoK, I'm not sure a compile-time solution is appropriate here.  There's nothing wrong with declaring and defining the H5-timing functions--it's just that sometimes you wants to *call* the timing/registration functions and other times you don't.  That suggests a runtime configuration option.